### PR TITLE
[Auth restructure 4] Fix examples/tests/docs and CI

### PR DESCRIPTION
**Pull request by marioortizmanero**
Part 4 of #207's split up. This fixes many other things related to the rewrite: the examples, tests, documentation, and CI.

The main library should pass the tests & CI, and the examples should work after this PR.

### CODE REVIEW

**Comment by marioortizmanero** [comment]
I'm currently updating the changelog to include a small guide on how to upgrade to the new version. I've opened the #218 issue for those with questions regarding the update, and I'll mention said issue in the changelog as well.

**Comment by ramsayleung** [comment]
Is there any required order for this [Auth restructure part N] PR to merge? Does one PR reply to another? If not, is it fine to merge individually when one PR review approves and CI pass?

**Comment by marioortizmanero** [comment]
You can merge as you like, GitHub should handle it correctly. Though the safest way to go is first 4, then 3, then 2, then 1, then separate-crates.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,94 @@
...
+* We've **renamed** a few structs and endpoints. The new names are quite similar, so the Rust compiler should suggest you what to change after an error. The only one you might not notice is the **environmental variables**: they're now `RSPOTIFY_CLIENT_ID`, `RSPOTIFY_CLIENT_SECRET` and `RSPOTIFY_REDIRECT_URI` to avoid collisions with other libraries.
+* We always use **`Option<T>`** for optional parameters now. This means that you might have to add `Some(...)` to some of your parameters. We were using both `Into<Option<T>>` and `Option<T>` but decided that either of these would be best as long as it's *consistent*. `Option<T>` has less magic, so we went for that one.
+* The core library has been split up with **features**. If you need `dotenv` just activate `env-file`, and if you need CLI functionality (`prompt_for_token` and similars), activate `cli`.
```
[the remark this answers is not in the document]
Comment (marioortizmanero) [reply CHANGELOG.md:20]: What's wrong with it? Similar as in close, or alike.

[the remark this answers is not in the document]
Comment (ramsayleung) [reply CHANGELOG.md:20]: It doesn't really matter.

**Review by ramsayleung — APPROVED** [review]

**Comment by ramsayleung** [comment]
Looks good to me, merged :)

**Pull request merged**